### ISSUE TITLE
LIBS: clock_gettime used unconditionally on linux for sdl2

### ISSUE THREAD

**Issue by mgerhardy** [mgerhardy/caveexpress#53]
HAVE_CLOCK_GETTIME is always set to 1 in the linux config header for sdl2

**Comment by mgerhardy** [comment]
```
Linking CXX executable ../../../../jsonconvert

../../../lib/libsdl2.a(SDL_syscond.c.o): In function `SDL_CondWaitTimeout_REAL':

/home/travis/build/mgerhardy/caveexpress/src/libs/sdl2/src/thread/pthread/SDL_syscond.c:112: undefined reference to `clock_gettime'

../../../lib/libsdl2.a(SDL_syssem.c.o): In function `SDL_SemWaitTimeout_REAL':

/home/travis/build/mgerhardy/caveexpress/src/libs/sdl2/src/thread/pthread/SDL_syssem.c:135: undefined reference to `clock_gettime'

../../../lib/libsdl2.a(SDL_systimer.c.o): In function `SDL_TicksInit':

/home/travis/build/mgerhardy/caveexpress/src/libs/sdl2/src/timer/unix/SDL_systimer.c:81: undefined reference to `clock_gettime'

../../../lib/libsdl2.a(SDL_systimer.c.o): In function `SDL_GetTicks_REAL':

/home/travis/build/mgerhardy/caveexpress/src/libs/sdl2/src/timer/unix/SDL_systimer.c:113: undefined reference to `clock_gettime'

../../../lib/libsdl2.a(SDL_systimer.c.o): In function `SDL_GetPerformanceCounter_REAL':

/home/travis/build/mgerhardy/caveexpress/src/libs/sdl2/src/timer/unix/SDL_systimer.c:144: undefined reference to `clock_gettime'

collect2: error: ld returned 1 exit status
```

**Comment by mgerhardy** [comment]
but only for the tools....

**Comment by mgerhardy** [comment]
fixed by 3714731bed2fbedcf4413da1bd32aae9f22247f1 and the following commits
